Xamarin.Forms.Core/ITemplatedItemsList.cs

```diff
@@ -0,0 +1,28 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.ComponentModel;
+using System.Collections;
+
+namespace Xamarin.Forms
+{
+	public interface ITemplatedItemsList<TItem> : IReadOnlyList<TItem>, INotifyCollectionChanged where TItem : BindableObject
```

Comment:
@jassmith I recall customers wanting access to the TIL and those folks will likely take a dependency on this interface. Are there any backwards compat caveats we want to make those customers aware of with respect to deping on these interfaces?